src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -176,6 +196,307 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             return (int)(byte*)(index + 7);
         }
 
+        public static unsafe int IndexOfAny(ref byte searchSpace, byte value0, byte value1, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            uint uValue0 = value0; // Use uint for comparisions to avoid unnecessary 8->32 extensions
+            uint uValue1 = value1; // Use uint for comparisions to avoid unnecessary 8->32 extensions
+            IntPtr index = (IntPtr)0; // Use UIntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
+            IntPtr nLength = (IntPtr)(uint)length;
+#if !netstandard10
+            if (Vector.IsHardwareAccelerated && length >= Vector<byte>.Count * 2)
+            {
+                unchecked
+                {
+                    int unaligned = (int)(byte*)Unsafe.AsPointer(ref searchSpace) & (Vector<byte>.Count - 1);
+                    nLength = (IntPtr)(uint)unaligned;
```

Comment:
Isn't this still backwards? If the purpose is to fix alignment, then you need to re-align the data on Vector<byte>.Count boundary. 'unaligned' above ends up being the number of bytes of misalignment. That would mean you need to advance Vector<byte>.Count - unaligned to get the data back into alignment.